internal/tiltfile/tiltfile_state.go

```diff
@@ -727,13 +728,29 @@ func (s *tiltfileState) translateK8s(resources []*k8sResource) ([]model.Manifest
 // on the pod (b/c of how we assemble resources, this corresponds to the first image target).
 // We won't collect container info (including DeployInfo) on any subsequent containers
 // (i.e. subsequent image targets), so will never be able to LiveUpdate them.
-func (s *tiltfileState) checkForImpossibleLiveUpdates(iTargs []model.ImageTarget) {
-	if len(iTargs) <= 1 {
-		// nothing to worry about!
-		return
-	}
-	for _, iTarg := range iTargs[1:] {
-		if !iTarg.AnyFastBuildInfo().Empty() || !iTarg.AnyLiveUpdateInfo().Empty() {
+func (s *tiltfileState) checkForImpossibleLiveUpdates(m model.Manifest) {
+	seenDeployedImage := false
+	for _, iTarg := range m.ImageTargets {
+		isDeployed := m.IsImageDeployed(iTarg)
+		isFirstDeployedImage := false
+		if isDeployed && !seenDeployedImage {
+			isFirstDeployedImage = true
+			seenDeployedImage = true
+		}
+
+		// This check only applies to images with live updates.
+		isInPlaceUpdate := !iTarg.AnyFastBuildInfo().Empty() || !iTarg.AnyLiveUpdateInfo().Empty()
+		if !isInPlaceUpdate {
+			continue
+		}
+
+		// TODO(nick): If an undeployed base image has a live-update component, we
+		// should probably emit a different kind of warning.
+		if !isDeployed {
+			continue
+		}
+
+		if !isFirstDeployedImage {
```

Comment:
hmm we probably don't want this whoooole error string for EVERY subsequent liveUpdated image -- maybe either `break` out after we find one impossible liveUpdate, or collect the image names and drop them into a single warning

Reply:
done!